src/main/java/frc/robot/subsystems/vision/PhotonVision.java

```diff
@@ -180,36 +180,33 @@ public EstimationRejectionReason shouldRejectEstimation(
             // TODO: do we always need to reject immediately here? maybe we can still use the next estimation even
             //  if the last estimation had no timestamp or was very close
-            return EstimationRejectionReason.LAST_ESTIMATED_POSE_TIMESTAMP_INVALID_OR_TOO_CLOSE;
-        }
-
-        final Pose2d nextEstimatedPosition2d = nextEstimatedPosition.toPose2d();
-        final Pose2d lastEstimatedPosition2d = lastEstimatedRobotPose.estimatedPose.toPose2d();
-        final Twist2d twist2dToNewEstimation = lastEstimatedPosition2d.log(nextEstimatedPosition2d);
-
-        final double xVel = twist2dToNewEstimation.dx / secondsSinceLastUpdate;
-        final double yVel = twist2dToNewEstimation.dy / secondsSinceLastUpdate;
-        final double thetaVel = twist2dToNewEstimation.dtheta / secondsSinceLastUpdate;
-        final double translationVel = Math.hypot(xVel, yVel);
-
-        Logger.recordOutput(
-                PhotonLogKey + "/TranslationVel", translationVel
-        );
-        Logger.recordOutput(
-                PhotonLogKey + "/ThetaVel", thetaVel
-        );
-
-        // assume hypot is positive (>= 0)
-        if ((translationVel >= maxLinearVelocity + PhotonVision.TranslationalVelocityTolerance)
-                || (Math.abs(thetaVel) >= maxAngularVelocity + PhotonVision.AngularVelocityTolerance)) {
-            // reject sudden pose changes resulting in an impossible velocity (cannot reach)
-            return EstimationRejectionReason.POSE_IMPOSSIBLE_VELOCITY;
+//            return EstimationRejectionReason.LAST_ESTIMATED_POSE_TIMESTAMP_INVALID_OR_TOO_CLOSE;
+//        }
+
+        // Only try calculating this rejection strategy if time > 0
+        if (secondsSinceLastUpdate > 0) {
+            final Pose2d nextEstimatedPosition2d = nextEstimatedPosition.toPose2d();
+            final Pose2d lastEstimatedPosition2d = lastEstimatedRobotPose.estimatedPose.toPose2d();
+            final Twist2d twist2dToNewEstimation = lastEstimatedPosition2d.log(nextEstimatedPosition2d);
+
+            final double xVel = twist2dToNewEstimation.dx / secondsSinceLastUpdate;
+            final double yVel = twist2dToNewEstimation.dy / secondsSinceLastUpdate;
+            final double thetaVel = twist2dToNewEstimation.dtheta / secondsSinceLastUpdate;
+            final double translationVel = Math.hypot(xVel, yVel);
+
+            Logger.recordOutput(PhotonLogKey + "/Rejection/TranslationVel", translationVel);
+            Logger.recordOutput(PhotonLogKey + "/Rejection/ThetaVel", thetaVel);
+
+            if ((Math.abs(translationVel) >= maxLinearVelocity + PhotonVision.TranslationalVelocityTolerance)
+                    || (Math.abs(thetaVel) >= maxAngularVelocity + PhotonVision.AngularVelocityTolerance)) {
+                // reject sudden pose changes resulting in an impossible velocity (cannot reach)
+                return EstimationRejectionReason.POSE_IMPOSSIBLE_VELOCITY;
+            }
```

Comment:
Honestly not sure what to do with this stuff, I removed one of the rejections at comp because I saw that most rejections had `LAST_ESTIMATED_POSE_TIMESTAMP_INVALID_OR_TOO_CLOSE` as the reason which didn't seem right. Probably needs more debugging/investigation into it.

Any ideas?

Reply:
Left the original rejection out for now, but threw another TODO on there.